server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/jobs/SqlLogStateJob.java

```diff
@@ -0,0 +1,104 @@
+import org.jkiss.dbeaver.model.exec.output.DBCServerOutputReader;
+import org.jkiss.dbeaver.model.runtime.AbstractJob;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.websocket.event.WSServerOutputLogInfo;
+import org.jkiss.dbeaver.model.websocket.event.session.WSLogSenderEvent;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class SqlLogStateJob extends AbstractJob {
+
+    private static final Log log = Log.getLog(SqlLogStateJob.class);
+
+    private final WebSession webSession;
+    private final DBCExecutionContext dbcExecutionContext;
+    private final DBCStatement dbcStatement;
+    private final DBCServerOutputReader dbcServerOutputReader;
+
+    public SqlLogStateJob(WebSession webSession, DBCExecutionContext dbcExecutionContext, DBCStatement dbcStatement,
+                          DBCServerOutputReader dbcServerOutputReader) {
+        super("Sql log state job");
+        this.webSession = webSession;
+        this.dbcExecutionContext = dbcExecutionContext;
+        this.dbcStatement = dbcStatement;
+        this.dbcServerOutputReader = dbcServerOutputReader;
+    }
+
+    @Override
+    protected IStatus run(DBRProgressMonitor monitor) {
+        if (!DBWorkbench.getPlatform().isShuttingDown()) {
+            try {
+                if (!dbcStatement.isStatementClosed()) {
+                    dumpOutput(monitor);
+                    schedule(100);
+                }
+            } catch (Exception e) {
+                log.debug(e);
+            }
+        }
+        return Status.OK_STATUS;
+    }
+
+    private void dumpOutput(DBRProgressMonitor monitor) {
+        if (!monitor.isCanceled()) {
+            DBCServerOutputReader outputReader = dbcServerOutputReader;
+            final DBCExecutionContext executionContext = dbcExecutionContext;
+
+            if (outputReader != null && outputReader.isAsyncOutputReadSupported() && executionContext != null) {
+                try {
+                    if (dbcStatement != null && !dbcStatement.isStatementClosed()) {
+                        List<WSServerOutputLogInfo> messages = new ArrayList<>();
+                        outputReader.readServerOutput(monitor, executionContext, null, dbcStatement, new DBCOutputWriter() {
+                            @Override
+                            public void println(@Nullable DBCOutputSeverity severity, @Nullable String message) {
+                                messages.add(new WSServerOutputLogInfo(severity.getName(), message));
+                            }
+
+                            @Override
+                            public void flush() {
+                                messages.clear();
+                            }
+                        });
+                        webSession.addSessionEvent(new WSLogSenderEvent(
+                            String.valueOf(executionContext.getContextId()),
```

Comment:
context id != task id

Reply:
if it is more convenient to pass the context ID, you need to discuss this with the frontend